storage/badger/operation/events.go

```diff
@@ -3,11 +3,34 @@
 package operation
 
 import (
+	"fmt"
+
 	"github.com/dgraph-io/badger/v2"
 
 	"github.com/onflow/flow-go/model/flow"
 )
 
+// prefixes used to distinguish service events
+const (
+	ServiceEventPrefixSetup  = byte(1)
+	ServiceEventPrefixCommit = byte(2)
+)
+
+// serviceEventTypeToPrefix returns byte prefix for given service event type
+func serviceEventTypeToPrefix(eventType string) byte {
```

Comment:
My thinking was that this is a storage layer which shouldn't really be concerned with validation. Should we maybe add extra validation step before?

Reply:
Yeah, I see where you're coming from, I agree the storage layer shouldn't be responsible for this kind of validation in general. But I think it should also be reasonably safe against panics. In this case validation is partly coupled to safety against crashes. 

We should definitely add a validation step before the event reaches this point, otherwise this will be guaranteed to represent a vulnerability to crashes. My point is more about approaching this defensively in general, so that even if one code-path leading to this function fails to implement proper validation, we can handle the resulting failure more gracefully than a panic. 

I think a reasonable general rule of thumb is, if the input which might cause a panic (if unvalidated) is expected to come from an external source, then we should use errors, and if the input is expected to come from an internal source, then it's safe to use panics at a lower level like this example.

Reply:
Yes, you are right.